.github/workflows/test-e2e.yml

```
@@ -51,6 +51,26 @@ jobs:
       - name: Vault Compatability
         run: make integration-test
       - run: make e2e-test
+
+  test-e2e-consul:
+    runs-on: ${{ endsWith(github.repository, '-enterprise') && fromJSON('["self-hosted", "ondemand", "linux"]') || 'ubuntu-latest' }}
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - uses: ./.github/actions/vault-secrets
+        with:
+          paths: |-
+            kv/data/github/hashicorp/nomad-enterprise/gha ELEVATED_GITHUB_TOKEN ;
+      - name: Git config token
+        if: endsWith(github.repository, '-enterprise')
+        run: git config --global url.'https://${{ env.ELEVATED_GITHUB_TOKEN }}@github.com'.insteadOf 'https://github.com'
```

I believe this should do the trick:

```suggestion
    runs-on: 'ubuntu-latest' # this job requires sudo, so not currently suitable for self-hosted runners
    steps:
      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
      - name: Git config token
        if: endsWith(github.repository, '-enterprise')
        run: git config --global url.'https://${{ secrets.ELEVATED_GITHUB_TOKEN }}@github.com'.insteadOf 'https://github.com'
```

other jobs in this predicament only set Git config token over in ent ([like this one](https://github.com/hashicorp/nomad-enterprise/blob/v1.6.2%2Bent/.github/workflows/test-core.yaml#L82-L83)), but I think the condition here will work fine.

---

Thanks! That also catches that we should pin the OS version for the sake of reliability.